File: game_plugin_interface/src/sc/protocol/responses/ObservationResponse.java
package sc.protocol.responses;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamAsAttribute;

@XStreamAlias(value="observed")
public class ObservationResponse
{
	@XStreamAsAttribute
	private String	roomId;
	
	public ObservationResponse(String roomId)
	{
		this.roomId = roomId;
	}

	public String getRoomId()
	{
		return this.roomId;
	}
}

File: sdk/src/server-api/sc/protocol/responses/ObservationResponse.kt
package sc.protocol.responses

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute
import sc.protocol.ResponsePacket

/** Sent to client as response to successfully joining a GameRoom as Observer. */
@XStreamAlias(value = "observed")
data class ObservationResponse(
        @XStreamAsAttribute
        val roomId: String
): ResponsePacket
